datascrubber/cli.py

```diff
@@ -0,0 +1,300 @@
+import logging
+import logging.handlers
+import argparse
+import os
+import sys
+import threading
+import traceback
+import boto3
+
+from . import ScrubWorkspaceInstance, RdsSnapshotFinder
+from .task_managers import Mysql, Postgresql
+
+
+def main():
+    args = parse_arguments()
+    configure_logging(args.log_mode, args.log_level)
+    logger = logging.getLogger()
+    logger.info('Starting up')
+
+    threads = []
+
+    if args.mysql_snapshots is not None:
+        for snap_id in args.mysql_snapshots:
+            thread = threading.Thread(
+                target=worker,
+                kwargs=({
+                    'dbms': 'mysql',
+                    'snapshot': snap_id,
+                    'target_accounts': args.share_with,
+                }),
+            )
+            threads.append(thread)
+
+    elif args.mysql_instances is not None:
+        for instance_id in args.mysql_instances:
+            thread = threading.Thread(
+                target=worker,
+                kwargs=({
+                    'dbms': 'mysql',
+                    'instance': instance_id,
+                    'target_accounts': args.share_with,
+                })
+            )
+            threads.append(thread)
+
+    elif args.mysql_hosts is not None:
+        for host in args.mysql_hosts:
+            thread = threading.Thread(
+                target=worker,
+                kwargs=({
+                    'dbms': 'mysql',
+                    'hostname': host,
+                    'target_accounts': args.share_with,
+                })
+            )
+            threads.append(thread)
+
+    if args.postgresql_snapshots is not None:
+        for snap_id in args.postgresql_snapshots:
+            thread = threading.Thread(
+                target=worker,
+                kwargs=({
+                    'dbms': 'postgresql',
+                    'snapshot': snap_id,
+                    'target_accounts': args.share_with,
+                }),
+            )
+            threads.append(thread)
+
+    elif args.postgresql_instances is not None:
+        for instance_id in args.postgresql_instances:
+            thread = threading.Thread(
+                target=worker,
+                kwargs=({
+                    'dbms': 'postgresql',
+                    'instance': instance_id,
+                    'target_accounts': args.share_with,
+                })
+            )
+            threads.append(thread)
+
+    elif args.postgresql_hosts is not None:
+        for host in args.postgresql_hosts:
+            thread = threading.Thread(
+                target=worker,
+                kwargs=({
+                    'dbms': 'postgresql',
+                    'hostname': host,
+                    'target_accounts': args.share_with,
+                })
+            )
+            threads.append(thread)
+
+    for thread in threads:
+        thread.start()
+
+    for thread in threads:
+        thread.join()
```

Comment:
Why not do the `start` and `join` in the same `for` loop?

Reply:
It would make the threads execute sequentially rather than in parallel, thus losing the benefits of threading.

start.....join, start....join etc.